dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java

```diff
@@ -121,6 +128,10 @@ public MetadataResource(SessionFactory sessionFactory, DockstoreWebserviceConfig
         this.config = config;
         this.bioWorkflowDAO = new BioWorkflowDAO(sessionFactory);
         this.serviceDAO = new ServiceDAO(sessionFactory);
+
+        Optional<StateListenerInterface> first = PublicStateManager.getInstance().getListeners().stream()
```

Comment:
This seems slightly fragile in that it relies on the listener having been added  to PublicStateManager before the MetadataResource is instantiated in DockstoreWebserviceApplication. Which it is now, and probably won't change, but still...

1. You could create and add the listener here instead of in DockstoreWebserviceApplication. The downside is that in theory you might have to deal with concurrency, but you could probably skip.
2. You could create a static instance of the listener in this class and add it from DockstoreWebserviceApplication
3. You could just leave it as is. With an additional optional throwing of an exception if it's null that brings down the web service.
4. You could just leave it as is

I lean towards 3, but am OK with 4.